addons/pos_self_order/demo/pos_restaurant_demo.xml

```diff
@@ -1,6 +1,11 @@
 <?xml version="1.0"?>
 <odoo>
-    <record id="pos_restaurant.pos_config_restaurant" model="pos.config">
-        <field name="self_order_view_mode">True</field>
-    </record>
+    <data noupdate="1">
+        <record id="pos_restaurant.pos_config_restaurant" model="pos.config">
+            <field name="self_order_view_mode">True</field>
+            <field name="self_order_table_mode">True</field>
+        </record>
+        <function model="restaurant.table" name="set_access_token_to_demo_data_records" />
+        <function model="pos.config" name="set_default_image" />
```

Comment:
Don't expose a procedure (method) that will only be used for creating demo. I think it's better to be more explicit here and specify the demo records you want to augment.